pkg/interface/src/views/components/ImageInput.tsx

```diff
@@ -55,25 +97,21 @@ export function ImageInput(props: ImageInputProps): ReactElement {
           {caption}
         </Label>
       ) : null}
-      <Row mt="2" alignItems="flex-end">
+      <Row mt="2" alignItems="flex-end" position='relative'>
+        {!field.value && !uploading && meta.error === undefined ? (<Prompt />) : null}
+        {uploading && meta.error === undefined ? (<Uploading />) : null}
```

Comment:
Calling closures as components is bad, because components are ideally referentially stable. Because referential equality is used in react's diffing algorithm, this causes the DOM elements to be removed and readded *on every render*. It's better to just call them as a regular function, as they won't be inserted into the component tree and will have no effect on the diffing. e.g. `{errorRetry()}`

Reply:
Thanks so much for this explanation, will refactor into regular functions.

Reply:
This is still an issue, because even though it looks like we aren't capturing any variables, you're still defining these components inside the scope of the component. You can either pull these out so that they're defined in the top level of the file, or just call them like `{errorRetry()}`